### PR TITLE
Cosmetic changes to install-service-worker

### PR DESCRIPTION
**Pull request by sebastianbenz**
* disable pre-cache verbose mode
* add links to sw docs
* add links to supported browsers
* some rephrasing

### CODE REVIEW

**Comment by sebastianbenz** [comment]
@ymotongpoo @jmadler PTAL

**Comment by ymotongpoo** [comment]
> disable pre-cache verbose mode
LGTM

> some rephrasing
I think [Service Worker](https://www.w3.org/TR/service-workers/) is a proper noun and camel case is preferred as described in the official spec.

> add links to sw docs
> add links to supported browsers

**Comment by sebastianbenz** [comment]
The official spec actually writes sw lower case as well.

The links are already there:

> Service Workers are currently available in [Chrome, Firefox and Opera](https://jakearchibald.github.io/isserviceworkerready/).

and 

> [Service workers](https://github.com/slightlyoff/ServiceWorker/blob/master/explainer.md)

**Comment by ymotongpoo** [comment]
Confirmed all points you mentioned. LGTM.

**Comment by sebastianbenz** [comment]
@jmadler can also please also take a look. Thanks!

File: src/20_Components/amp-install-serviceworker.html [hunk 1]
```diff
@@ -24,24 +26,28 @@
...
-  The reason why the latter option is required is because browsers can't install Service Worker from other domains for security reasons. For details of the security considerations of Service Worker, [see official specs](https://www.w3.org/TR/service-workers/#security-considerations).
+  `amp-install-serviceworker` should have both properties `src` and `data-iframe-src` set; `src` is the location of service worker to be installed when the original AMP pages is accessed, `data-iframe-src` is the location of HTML file which calls service worker installation process when the cached AMP pages is accessed.
+  The reason why the latter option is required is because browsers can't install service worker from other domains for security reasons. For details of the security considerations of service worker, [see official specs](https://www.w3.org/TR/service-workers/#security-considerations).
+
+  One thing to note: the service worker will only be installed if the page is open longer than 20s.
```
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: @ithinkihaveacat FYI

**Comment by sebastianbenz** [comment]
@jmadler Friendly Ping

File: src/20_Components/amp-install-serviceworker.html [hunk 2]
```diff
@@ -2,10 +2,12 @@
   #### Introduction
 
   The [amp-install-serviceworker](https://github.com/ampproject/amphtml/blob/master/extensions%2Famp-install-serviceworker%2Famp-install-serviceworker.md)
-  component enables Service Worker installation via original and cached AMP pages.
+  component enables service worker installation via original and cached AMP pages.
```
Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: I would suggest "same- and cross-origin" instead of "original and cached"

File: src/20_Components/amp-install-serviceworker.html [hunk 3]
```diff
@@ -2,10 +2,12 @@
...
   
-  The benefit of Service Worker is its capability, such as runnning on background even when the browser is closed, caching required contents in advance of rendering time and sending notifications to users and so on.
-  This demonstration installs a Service Worker that caches this demo page using Cache API triggered by the Service Worker on background. You can confirm that this demo works even offline once the Service Worker gets installed.
+  [Service workers](https://github.com/slightlyoff/ServiceWorker/blob/master/explainer.md) can run in the background even when the browser is closed, cache required contents in advance of rendering time and send notifications to users and more.
```
Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: "in advance of rendering time, send notifications to users, and more"

File: src/20_Components/amp-install-serviceworker.html [hunk 4]
```diff
@@ -24,24 +26,28 @@
 </head>
 <body>
 
-  <!-- #### Basic Usage -->
+  <!-- #### Installing the Service Worker -->
   <!--
-  amp-install-serviceworker should have both properties `src` and `data-iframe-src`; `src` is the location of Service Worker to be installed when the original AMP pages is accessed, `data-iframe-src` is the location of HTML file which calls Service Worker installation process when the cached AMP pages is accessed.
-  The reason why the latter option is required is because browsers can't install Service Worker from other domains for security reasons. For details of the security considerations of Service Worker, [see official specs](https://www.w3.org/TR/service-workers/#security-considerations).
+  `amp-install-serviceworker` should have both properties `src` and `data-iframe-src` set; `src` is the location of service worker to be installed when the original AMP pages is accessed, `data-iframe-src` is the location of HTML file which calls service worker installation process when the cached AMP pages is accessed.
```
Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: data-iframe-src is optional, but we recommend including it.  I'd reword a bit -- something like:

Calls to `amp-install-serviceworker` can include the `src` and `data-iframe-src` attributes.  `src` is the location of the service worker to be installed, and is used when the document is accessed via the same origin as the service worker file.  When the document is accessed via a different origin than the one the service worker is scoped for, the HTML document noted in the `data-iframe-src`  attribute is used to install the service worker.

File: src/20_Components/amp-install-serviceworker.html [hunk 5]
```diff
@@ -24,24 +26,28 @@
...
+  The reason why the latter option is required is because browsers can't install service worker from other domains for security reasons. For details of the security considerations of service worker, [see official specs](https://www.w3.org/TR/service-workers/#security-considerations).
```
Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: Might make sense to explicitly call out the Google AMP Cache as a cross-origin access point.

File: src/20_Components/amp-install-serviceworker.html [hunk 6]
```diff
@@ -24,24 +26,28 @@
...
       src="/sw.js"
       data-iframe-src="https://ampbyexample.com/sw.html"
       layout="nodisplay">
   </amp-install-serviceworker>
   <!--
-  <p class="card warning">`sw.js` here is generated using [sw-precache](https://github.com/GoogleChrome/sw-precache), which generates a [Service Worker JavaScript file](https://ampbyexample.com/sw.js) that caches specified static files. See example in the repository of `sw-precache` and [gulpfile.js](https://github.com/ampproject/amp-by-example/blob/master/gulpfile.js) for how to use it.</p> In this sample, when users access this AMP page for the 2nd time, the Service Worker is installed and start caching all image files and video files under root path of this domain.
+  <p class="card warning">The service worker in this sample ([sw.js](https://ampbyexample.com/sw.js)) is generated using [sw-precache](https://github.com/GoogleChrome/sw-precache). With sw-precache you can generate service worker code that will precache specific resources.</p> 
+  
+  In this sample, when users access this AMP page for the 2nd time, the service worker is installed and start caching all image files and video files under root path of this domain.
```
Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: style nit: I'd use "second"

Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: s/start caching/caches/
s/under root path/under the root path/

File: src/20_Components/amp-install-serviceworker.html [hunk 7]
```diff
@@ -24,24 +26,28 @@
...
   
-  In order to confirm if the Service Worker is installed and it is working as expected, follow the procedure below:
+  In order to confirm if the service worker is installed and it is working as expected, follow the procedure below:
   1. Access to https://ampbyexample.com/components/amp-install-serviceworker/ twice.
```
Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: remove "to"

File: src/20_Components/amp-install-serviceworker.html [hunk 8]
```diff
@@ -24,24 +26,28 @@
...
-  2. Open Chrome Dev Tools and navigate to "Resources" tab. You may see "Service Workers" section and see `/sw.js` is installed.
-  3. Then open "Cached Storage". You'll find a lot of files whose names start with "sw-precache-v1--", which are cached images and videos.
-  4. Disable your network connection and reload this page. It still works! On "Network" tab in Dev Tools, you'll find "(from ServiceWorker)" notation in the rows of image/video file fetch events.  
+  2. Open Chrome Dev Tools and navigate to "Resources" tab. In the "Service Workers" section you can see that `/sw.js` is installed.
```
Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: to the "Resources" tab

File: src/20_Components/amp-install-serviceworker.html [hunk 9]
```diff
@@ -24,24 +26,28 @@
...
+  3. Then open "Cached Storage". You'll find a lot of files whose names start with `sw-precache-v1--`, these are the cached images and videos.
```
Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: the "Cached Storage" tab

File: src/20_Components/amp-install-serviceworker.html [hunk 10]
```diff
@@ -24,24 +26,28 @@
...
+  4. Disable your network connection and reload this page. It still works! In the "Network" tab in Dev Tools, you'll find see which resources are cached by the service worker.
```
Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: remove "find"

File: src/20_Components/amp-install-serviceworker.html [hunk 11]
```diff
@@ -24,24 +26,28 @@
...
+  In order to confirm if the service worker is installed and it is working as expected, follow the procedure below:
   1. Access to https://ampbyexample.com/components/amp-install-serviceworker/ twice.
-  2. Open Chrome Dev Tools and navigate to "Resources" tab. You may see "Service Workers" section and see `/sw.js` is installed.
-  3. Then open "Cached Storage". You'll find a lot of files whose names start with "sw-precache-v1--", which are cached images and videos.
-  4. Disable your network connection and reload this page. It still works! On "Network" tab in Dev Tools, you'll find "(from ServiceWorker)" notation in the rows of image/video file fetch events.  
+  2. Open Chrome Dev Tools and navigate to "Resources" tab. In the "Service Workers" section you can see that `/sw.js` is installed.
+  3. Then open "Cached Storage". You'll find a lot of files whose names start with `sw-precache-v1--`, these are the cached images and videos.
+  4. Disable your network connection and reload this page. It still works! In the "Network" tab in Dev Tools, you'll find see which resources are cached by the service worker. 
   -->
```
Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: Might make sense to add a note to some blogpost that explains how SWs can be used for things other than caching docs for offline access.

**Comment by jmadler** [comment]
Comments added

File: src/20_Components/amp-install-serviceworker.html (hunk 2 above)
Comment (ithinkihaveacat) [inline src/20_Components/amp-install-serviceworker.html]: Not sure if this is 100% correct right now--the iframe install is gated on a call to [`isProxyOrigin()`](https://github.com/ampproject/amphtml/blob/master/extensions/amp-install-serviceworker/0.1/amp-install-serviceworker.js#L45) which in turn is [hard-coded to `cdn.ampproject.org`](https://github.com/ampproject/amphtml/blob/master/src/url.js#L238). I haven't traced through exactly what happens, but there's definitely some differences in behavior depending on whether the document is on or off the cache. (Not just same versus cross-origin.)

Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: Good point.  Would "same- and (in some cases) cross-origin" work?

Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: I went for "same origin or via the Google AMP Cache."

File: src/20_Components/amp-install-serviceworker.html (hunk 3 above)
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: oxford comma FTW. Done.

File: src/20_Components/amp-install-serviceworker.html (hunk 4 above)
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: done. thanks!

File: src/20_Components/amp-install-serviceworker.html (hunk 5 above)
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: done

File: src/20_Components/amp-install-serviceworker.html (hunk 6 above)
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: all three done

File: src/20_Components/amp-install-serviceworker.html (hunk 7 above)
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: rephrased the whole sentence

File: src/20_Components/amp-install-serviceworker.html (hunk 8 above)
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: done

File: src/20_Components/amp-install-serviceworker.html (hunk 9 above)
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: done

File: src/20_Components/amp-install-serviceworker.html (hunk 10 above)
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: done

File: src/20_Components/amp-install-serviceworker.html (hunk 11 above)
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: What's a good post?

**Comment by sebastianbenz** [comment]
Thanks Jordan and Michael. @jmadler @ithinkihaveacat PAL

File: src/20_Components/amp-install-serviceworker.html (hunk 11 above)
Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: Maybe this one: https://github.com/slightlyoff/ServiceWorker/blob/master/explainer.md

**Comment by jmadler** [comment]
Can you push the updated commit?

File: src/20_Components/amp-install-serviceworker.html (hunk 11 above)
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: I added this already.

**Comment by sebastianbenz** [comment]
@jmadler done - I thought I did already.

**Comment by sebastianbenz** [comment]
PTAL

File: src/20_Components/amp-install-serviceworker.html [hunk 12]
```diff
@@ -24,24 +26,27 @@
...
-  The reason why the latter option is required is because browsers can't install Service Worker from other domains for security reasons. For details of the security considerations of Service Worker, [see official specs](https://www.w3.org/TR/service-workers/#security-considerations).
+Calls to amp-install-serviceworker can include the `src` and `data-iframe-src` attributes. `src` is the location of the service worker to be installed, and is used when the document is accessed via the same origin as the service worker file. When the document is accessed via the Google AMP Cache, the HTML document noted in the `data-iframe-src` attribute is used to install the service worker. We recommend using both attributes.
+
+  The reason why the `data-iframe-src` option is required is because browsers can't install service worker from other domains for security reasons. For details of the security considerations of service worker, [see official specs](https://www.w3.org/TR/service-workers/#security-considerations).
```
Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: data-iframe-src isn't required, it's recommended

File: src/20_Components/amp-install-serviceworker.html [hunk 13]
```diff
@@ -24,24 +26,27 @@
...
+  <p class="card warning">The service worker in this sample ([sw.js](https://ampbyexample.com/sw.js)) is generated using [sw-precache](https://github.com/GoogleChrome/sw-precache). With sw-precache you can generate service worker code that will precache specific resources.</p> 
+  
+  In this sample, when users access this AMP page for the second time, the service worker is installed and caches all image files and video files under the root path of this domain.
   
-  In order to confirm if the Service Worker is installed and it is working as expected, follow the procedure below:
+  To confirm that the service worker is installed and works as expected, follow the following steps:
   1. Access to https://ampbyexample.com/components/amp-install-serviceworker/ twice.
```
Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: remove 'to'

**Comment by jmadler** [comment]
Added a couple minor suggestions.  LGTM otherwise

File: src/20_Components/amp-install-serviceworker.html (hunk 12 above)
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: done

File: src/20_Components/amp-install-serviceworker.html (hunk 13 above)
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: done

**Comment by sebastianbenz** [comment]
Merged after applying Jordan's suggestions. Thanks everyone for the review!
